internal/bminventory/inventory.go

```diff
@@ -6158,6 +6174,178 @@ func (b *bareMetalInventory) V2UpdateHostIgnitionInternal(ctx context.Context, p
+		APIVersion string `yaml:"apiVersion"`
+		Kind       string `yaml:"kind"`
+		Metadata   struct {
+			Name string `yaml:"name"`
+		} `yaml:"metadata"`
+		Spec struct {
+			PullSecretRef *struct {
+				Name string `yaml:"name"`
+			} `yaml:"pullSecretRef,omitempty"`
+			SSHAuthorizedKey string `yaml:"sshAuthorizedKey,omitempty"`
+		} `yaml:"spec"`
+	}{
+		APIVersion: "agent-install.openshift.io/v1beta1",
+		Kind:       "InfraEnv",
+		Metadata: struct {
+			Name string `yaml:"name"`
+		}{
+			Name: *infraEnv.Name,
+		},
+		Spec: struct {
+			PullSecretRef *struct {
+				Name string `yaml:"name"`
+			} `yaml:"pullSecretRef,omitempty"`
+			SSHAuthorizedKey string `yaml:"sshAuthorizedKey,omitempty"`
+		}{
+			PullSecretRef: &struct {
+				Name string `yaml:"name"`
+			}{Name: "pull-secret"},
+			SSHAuthorizedKey: infraEnv.SSHAuthorizedKey,
+		},
+	}
+
+	infraEnvYAML, err := yaml.Marshal(infraEnvObj)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to marshal infraEnv YAML")
+	}
+	if err = os.WriteFile(filepath.Join(manifestsDir, "infraenv.yaml"), infraEnvYAML, 0600); err != nil {
+		return "", errors.Wrap(err, "failed to write infraenv.yaml")
+	}
+
+	pullSecretObj := struct {
+		APIVersion string `yaml:"apiVersion"`
+		Kind       string `yaml:"kind"`
+		Metadata   struct {
+			Name string `yaml:"name"`
+		} `yaml:"metadata"`
+		Type       string            `yaml:"type"`
+		StringData map[string]string `yaml:"stringData"`
+	}{
+		APIVersion: "v1",
+		Kind:       "Secret",
+		Type:       "kubernetes.io/dockerconfigjson",
+		StringData: map[string]string{
+			".dockerconfigjson": infraEnv.PullSecret,
+		},
+	}
+	pullSecretObj.Metadata.Name = "pull-secret"
+
+	pullSecretYAML, err := yaml.Marshal(pullSecretObj)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to marshal pull secret YAML")
+	}
+	if err = os.WriteFile(filepath.Join(manifestsDir, "pull-secret.yaml"), pullSecretYAML, 0600); err != nil {
+		return "", errors.Wrap(err, "failed to write pull-secret.yaml")
+	}
+
+	mirrorDir := filepath.Join(tempDir, "mirror")
+	if err = os.MkdirAll(mirrorDir, 0755); err != nil {
+		return "", errors.Wrap(err, "failed to create mirror directory")
+	}
+
+	if err = os.WriteFile(filepath.Join(mirrorDir, "registries.conf"), []byte(getOVERegistriesConf()), 0600); err != nil {
+		return "", errors.Wrap(err, "failed to write registries.conf")
+	}
+
+	releaseImage, err := b.versionsHandler.GetReleaseImage(ctx, infraEnv.OpenshiftVersion, infraEnv.CPUArchitecture, infraEnv.PullSecret)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get release image for version %s", infraEnv.OpenshiftVersion)
+	}
+
+	ocRelease := oc.NewRelease(
+		b.executer,
+		oc.Config{MaxTries: oc.DefaultTries, RetryDelay: oc.DefaltRetryDelay},
+		b.mirrorRegistriesConfig,
+		system.NewLocalSystemInfo(),
+	)
+	release, err := b.installerCache.Get(ctx, *releaseImage.URL, "", infraEnv.PullSecret, ocRelease, infraEnv.OpenshiftVersion, infraEnv.ClusterID)
```

Comment:
This is assuming that we have a release image configured that matches the infraenv version. That's not always the case.

Also `ClusterID` will not be set for late-binding cases. Is that something we need to consider here?
Also if we only need to consider cases where infraenv is assigned to a cluster it would be better to get the cluster version and use that to get the release image.